lib/que/worker.rb

```diff
@@ -165,13 +165,17 @@ def work
         @locker.with_locked_job do |job|
           return :job_not_found if job.nil?
 
+          klass = class_for(job[:job_class])
+          job_instance = klass.new(job)
```

Comment:
This class is now initialised outside of the duration metric but I don't think this will make a notable difference

Reply:
I'd be mildly wary of this because the application can potentially override the initialiser and we can't make any assumption on its duration. Also, the behaviour may change if the initialiser raises.

Can we instead pass the raw job attributes to the proc?

Reply:
What do you mean by passing raw attributes? Do you mean we need to drop the support for a proc and just use a static hash?

Reply:
Ok just to verify what you're suggesting: Wel'll modify the proc to accept the hash of attributes instead of an instance of the job itself? And then we can call `klass.log_context_proc(job)` [here](https://github.com/gocardless/que/pull/84/files#diff-8fd8f62218ddc6ebe0d868e1f9790fac8d3f96274cf3675b1200891432abd38aR178)?

Reply:
Good call - I updated it. We have slightly more limited capabilites now but I think it's a fair tradeoff.